### ISSUE TITLE
bug fixes

### ISSUE THREAD

**Issue by AbdelrahimKoura** [PacktPublishing/Hands-On-Generative-Adversarial-Networks-with-PyTorch-1.x#4]
In chapter 4, during the execution file dcgan.py in the training loop, invoke the criterion loss function with two parameters "criterion(y_real, real_label)" get error type long where an expected float

**Comment by MattTucker22689** [comment]
Here's the fix to that bug:
````       
        # Update D with real data
        netD.zero_grad()
        y_real = netD(x_real)
        real_label = real_label.float()
        loss_D_real = criterion(y_real, real_label)
        loss_D_real.backward()

        # Update D with fake data
        z_noise = torch.randn(x_real.size(0), Z_DIM, 1, 1, device=device)
        x_fake = netG(z_noise)
        y_fake = netD(x_fake.detach())
        fake_label = fake_label.float()
        loss_D_fake = criterion(y_fake, fake_label)
        loss_D_fake.backward()
        optimizerD.step()

        # Update G with fake data
        netG.zero_grad()
        y_fake_r = netD(x_fake)
        real_label = real_label.float()
        loss_G = criterion(y_fake_r, real_label)
        loss_G.backward()
        optimizerG.step()
````
        
Notice the lines `thing = thing.float()` right before each loss is calculated.

You may also run into a problem with the dataloader and the number of workers it's implementing. To solve that try using this `dataloader = torch.utils.data.DataLoader(dataset, batch_size=BATCH_SIZE, shuffle=True, num_workers=1)`

**Comment by MattTucker22689** [comment]
I should note on my dataloader solution I only changed `num_workers` to `1`
